CMakeLists.txt

```diff
@@ -379,8 +379,12 @@ else()
     message(STATUS "AES support enabled")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maes")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -maes")
-  elseif(ARM) #NB ARMv8 DOES support AES, but not yet coded
-    message(STATUS "AES support disabled (not available on ARM)")
+  elseif(ARM6 OR ARM7)
+    message(STATUS "AES support disabled (not available on ARMv6 or ARMv7)")
+  elseif(ARM8)
+    message(STATUS "AES support enabled for ARMv8-A")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+crypto")
```

Comment:
So that actually requires 2 lines - one to set the CPU features in this location, and then a second line later on (presumably within the larger 'ARM' section just beneath) to act on that CPU_FEATURES flag. 

I'm not sure what we're saving at this point by making the change until someone can code up the multiple nested checks required to select the correct flags for ARMv7. When that's done, sure.